packages/@aws-cdk/assert/lib/canonicalize-assets.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Reduce template to a normal form where asset references have been normalized
+ *
+ * This makes it possible to compare templates if all that's different between
+ * them is the hashes of the asset values.
+ *
+ * Currently only handles parameterized assets, but can (and should)
+ * be adapted to handle convention-mode assets as well when we start using
+ * more of those.
```

Comment:
Since this is eminently coming, can we just support that as well?

Reply:
I'd rather not right now.